### PR TITLE
Feat(Task-10): As a User, I want to be to see the weather reports of 5 days.

### PR DESCRIPTION
**Pull request by SamuFigueirido**

### CODE REVIEW

File: src/app/main-page/main-page.controller.js [hunk 1]
```diff
@@ -21,8 +24,65 @@ export default class MainPageController {
             .catch(response => {
                 console.log('There are no nearby cities');
             });
-        console.log('Cities: ', this.nearbyCities);
+        this.openWeatherMapsService.getCurrentWeather(this.param)
+            .then(response => {
+                console.log(response);
```
Comment (eloydotco) [inline src/app/main-page/main-page.controller.js]: remove this log

File: src/app/main-page/main-page.controller.js [hunk 2]
```diff
@@ -21,8 +24,65 @@ export default class MainPageController {
...
+                let cont = -1;
+                let name = response.city.name;
+                let country = response.city.country;
```
Comment (eloydotco) [inline src/app/main-page/main-page.controller.js]: name and country variables can be const. right?

File: src/app/main-page/main-page.controller.js [hunk 3]
```diff
@@ -21,8 +24,65 @@ export default class MainPageController {
...
+                response.list.forEach(element => {
+                    let flag = false;
+                    const day = new Date(element.dt * 1000).getDate();
+                    this.days.forEach(item => {
+                        if (item.day === day) {
```
Comment (eloydotco) [inline src/app/main-page/main-page.controller.js]: it's ok, but it would be nice a better variable names like:
day -> currentDay
item -> day
I think are more readable and intuitive, what do you think?

File: src/app/main-page/main-page.controller.js [hunk 4]
```diff
@@ -21,8 +24,65 @@ export default class MainPageController {
...
+
+                    if (!flag) {
+                        cont++;
+                        this.days.push({
+                            main: element.dt * 1000,
+                            day: day,
+                            dayTitle: value,
+                            city: {
+                                name: name,
+                                country: country,
```
Comment (eloydotco) [inline src/app/main-page/main-page.controller.js]: this can be written like this -> `city: {name, country}`
just a tip, it's not necessary to change :D

File: src/app/services/openWeatherMaps.service.js [hunk 5]
```diff
@@ -1,19 +1,40 @@
...
         const request = {
             method: 'GET',
-            url: `http://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}`,
+            url: `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${this.apiKey}`,
+            params: {
+                units: 'metric',
+                mode: 'json',
+            }
+        }
+        const self = this;
```
Comment (eloydotco) [inline src/app/services/openWeatherMaps.service.js:16]: why is this needed??

File: src/app/components/container-days/container-days.component.html [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<md-content class="tab-container">
+    <md-tabs layout="row" layout-align="center">
+        <md-tab label="{{day.dayTitle}}" ng-repeat="day in $ctrl.listDays track by $index">
+            <div class="day-container">
+                <div class="row title-date-country">
+                    <div class="date-title">
+                        {{day.main | date:'fullDate'}}
+                    </div>
+                    <a class="city-name">{{day.city.name}}, {{day.city.country}}</a>
```
Comment (eloydotco) [inline src/app/components/container-days/container-days.component.html]: Can these be one-time bindings??

File: src/app/components/container-days/container-days.component.html [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+                </div>
+                <div class="tile">
+                    <div class="row hour" ng-repeat="dayHour in day.hours track by $index">
+                        {{dayHour.hour | date:'shortTime'}}
+                        <div class="row weather">
+                            <div class="icon-temp">
+                                <img class="weather-icon" layout="row" layout-align="center center"
+                                    src='http://openweathermap.org/img/w/{{dayHour.weather.icon}}.png'>
```
Comment (gbtec-ext-jorgevillar) [inline src/app/components/container-days/container-days.component.html]: Use `ng-src` instead `src` to avoid possible digest cycle bugs

File: src/app/main-page/main-page.component.scss [hunk 8]
```diff
@@ -19,36 +20,52 @@ main-page {
...
     display: flex;
     align-items: center;
   }
+
+  .title-nearby-cities {
+    color: white;
+    text-shadow: $shadow-border;
+    font-size: 20px;
+    border-bottom: 2px;
+    padding-top: 10px;
```
Comment (gbtec-ext-jorgevillar) [inline src/app/main-page/main-page.component.scss]: You can set the padding in one line

File: src/app/components/logo-letters/logo-letters.component.scss [hunk 9]
```diff
@@ -31,34 +31,25 @@ start-page {
         padding-bottom: 25px;
     }
 
-    .title {
-        display: table;
-        cursor: pointer;
-    }
-
     h1 {
```
Comment (gbtec-ext-jorgevillar) [inline src/app/components/logo-letters/logo-letters.component.scss:34]: You already have `h1` selector above, join the styles

File: src/app/components/logo-letters/logo-letters.component.scss [hunk 10]
```diff
@@ -31,34 +31,25 @@ start-page {
...
         font-size: 40px;
     }
 
-    h1,
-    .reg {
-        text-shadow: 1px 1px 2px black;
-        display: table-cell;
-    }
-
     .reg {
```
Comment (gbtec-ext-jorgevillar) [inline src/app/components/logo-letters/logo-letters.component.scss:38]: You already have `.reg` selector above, join the styles

File: src/app/components/list-cities/list-cities.component.scss [hunk 11]
```diff
@@ -19,51 +28,59 @@ start-page {
     }
 
     li {
-        margin: 10px 25px;
+        &.list-cities {
+            margin: 10px 25px;
+            height: 19px;
+            background-color: rgba(0, 0, 0, 0.55);
+            margin-left: 16px;
+            margin-right: 16px;
```
Comment (gbtec-ext-jorgevillar) [inline src/app/components/list-cities/list-cities.component.scss]: You're overriding the margin style in the first line of this selector, you can do it like this -> `margin: 10px 16px;`

File: src/app/main-page/main-page.controller.js [hunk 12]
```diff
@@ -21,8 +24,65 @@ export default class MainPageController {
             .catch(response => {
                 console.log('There are no nearby cities');
             });
-        console.log('Cities: ', this.nearbyCities);
+        this.openWeatherMapsService.getCurrentWeather(this.param)
+            .then(response => {
```
Comment (eloydotco) [inline src/app/main-page/main-page.controller.js:29]: maybe it's a good idea to move this logic to an utils or service, what do you think??

**Review by eloydotco — CHANGES_REQUESTED** [review]

File: src/app/main-page/main-page.controller.js [hunk 13]
```diff
@@ -21,8 +24,65 @@ export default class MainPageController {
...
+
+                    const hourObject = {
+                        hour: (element.dt * 1000),
+                        temperature: {
+                            main: element.main.temp,
+                            temp_max: element.main.temp_max,
+                            temp_min: element.main.temp_min
+                        },
+                        humidity: element.main.humidity,
+                        weather: {
```
Comment (gbtec-ext-jorgevillar) [inline src/app/main-page/main-page.controller.js]: Here, I think you can simply do: 
`weather: element.weather[0]`

File: src/app/main-page/main-page.controller.js [hunk 14]
```diff
@@ -21,8 +24,65 @@ export default class MainPageController {
...
+                                name: name,
+                                country: country,
+                            },
+                            hours: []
+                        })
+                    };
+
+                    const hourObject = {
+                        hour: (element.dt * 1000),
+                        temperature: {
```
Comment (gbtec-ext-jorgevillar) [inline src/app/main-page/main-page.controller.js]: Here, I think you can simply do: 
`weather: element.main`

File: src/app/main-page/main-page.controller.js [hunk 15]
```diff
@@ -21,8 +24,65 @@ export default class MainPageController {
...
+                        }
+                    };
+                    this.days[cont].hours.push(hourObject);
+                });
+                this.days[0].dayTitle = 'TODAY';
+                this.days[1].dayTitle = 'TOMORROW';
+            })
+            // .catch(response => {
+            //     console.log('There is no weather for this city');
+            // });
```
Comment (gbtec-ext-jorgevillar) [inline src/app/main-page/main-page.controller.js]: Remove commented code

File: src/app/services/openWeatherMaps.service.js (hunk 5 above)
Comment (SamuFigueirido) [reply src/app/services/openWeatherMaps.service.js:16]: Jorge told me to put it there to return the response.

**Review by gbtec-ext-jorgevillar — CHANGES_REQUESTED** [review]

File: src/app/services/openWeatherMaps.service.js (hunk 5 above)
Comment (gbtec-ext-jorgevillar) [reply src/app/services/openWeatherMaps.service.js:16]: For some reason 'this' is not available inside the $https.then

File: src/app/main-page/main-page.controller.js (hunk 13 above)
Comment (SamuFigueirido) [reply src/app/main-page/main-page.controller.js]: But there is also the "id", and I dont't want it

**Review by eloydotco — APPROVED** [review]

**Review by gbtec-ext-jorgevillar — APPROVED** [review]
